From 3d0b2176e04261ab4ac095ff2a17db077fc1e46d Mon Sep 17 00:00:00 2001
From: Vinod Koul <vkoul@kernel.org>
Date: Fri, 22 Dec 2023 15:10:01 +0530
Subject: [PATCH] dmaengine: xilinx: xdma: statify xdma_prep_interleaved_dma

xdma_prep_interleaved_dma() was local to file but not declared static,
leading to warning:

drivers/dma/xilinx/xdma.c:729:1: warning: no previous prototype for 'xdma_prep_interleaved_dma' [-Wmissing-prototypes]
  729 | xdma_prep_interleaved_dma(struct dma_chan *chan

Reported-by: Stephen Rothwell <sfr@canb.auug.org.au>
Signed-off-by: Vinod Koul <vkoul@kernel.org>
Link: https://lore.kernel.org/r/20231222094001.731889-1-vkoul@kernel.org
Signed-off-by: Vinod Koul <vkoul@kernel.org>
---
Link: https://git.kernel.org/vkoul/dmaengine/c/3d0b2176e04261ab4ac095ff2a17db077fc1e46d
---
 drivers/dma/xilinx/xdma.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/dma/xilinx/xdma.c b/drivers/dma/xilinx/xdma.c
index 9360b85131ef..4ebc90b41bdb 100644
--- a/drivers/dma/xilinx/xdma.c
+++ b/drivers/dma/xilinx/xdma.c
@@ -725,7 +725,7 @@ xdma_prep_dma_cyclic(struct dma_chan *chan, dma_addr_t address,
  * @xt: DMA transfer template
  * @flags: tx flags
  */
-struct dma_async_tx_descriptor *
+static struct dma_async_tx_descriptor *
 xdma_prep_interleaved_dma(struct dma_chan *chan,
 			  struct dma_interleaved_template *xt,
 			  unsigned long flags)
-- 
2.43.0

